alibi_detect/cd/utils.py

```diff
@@ -98,3 +101,16 @@ def new_model_fn(x: np.ndarray) -> np.ndarray:
         return preds
 
     return new_model_fn
+
+
+def get_input_shape(shape: Union[Tuple, None], x_ref: Union[np.ndarray, list]) -> Union[Tuple, None]:
+    """ Optionally infer shape from reference data. """
```

Comment:
What is meant by "optionally" here?

Reply:
Yes, it simply factors out getting the shape for different detectors, so this instead of a few repeated lines in different inits.